railties/lib/rails/commands/app/update_command.rb

```diff
@@ -34,6 +34,8 @@ def bin
 
         desc "public_directory", "Add or update files in the application public/ directory", hide: true
         def public_directory
+          return if generator_options[:api]
```

Comment:
What do you think about moving this condition into `create_public_files`? I think this has two benefits
- (maybe) prevents needing `remove_public_files`?
- consistent with the `skip_active_storage?` condition in `update_active_storage`

Reply:
Currently, we call `create_public_files` even if an application is API-only mode. This is because to generate `public/robots.txt`. I think this is intended behavior. So we can't remove `remove_public_files` so far.

`update_active_storage` implemented before we have `update_command.rb`. So in my personal opinion, checking inside `update_command.rb` is better to simplify `AppGenerator`. But that is out of scope of this PR. So I moved the check inside `create_public_files` for consistently. Thanks.